highfidelity/hifi
# Entity property range/step audit

- Modify the _propertyStringsToEnums structure to instead be _propertyInfos with EntityPropertyInfo structs consisting of the property enum and a min and max
- Add alternate _WITH_RANGE versions of the ADD_PROPERTY_TO_MAP and ADD_GROUP_PROPERTY_TO_MAP macros to define selected numerical properties with min and max values
- Add getPropertyInfo to Entity API to access the EntityPropertyInfo struct by property name
- Lookup property ranges when Entity Properties is opened, and replace any undefined min/max values with range values that are defined in code
- Tweak/add/remove min/max/step/decimal values for many numerical properties in entityProperties.js, in some cases to adjust for the fact that they are now driven by draggable numbers instead of individual stepping

Addresses:
https://highfidelity.manuscript.com/f/cases/19992
https://highfidelity.manuscript.com/f/cases/19993
https://highfidelity.manuscript.com/f/cases/19546

Pre-Merge Test Plan:
- Enter Interface in desktop mode into any domain where you have edit permissions
- Open Create and create a Particle entity
- Verify the default Horizontal Angle range is now 0 Start to 180 Finish
- Verify if adjusting Size Middle value that Size Finish value changes with it after releasing
- Verify if adjusting Alpha Middle value that Alpha Finish value changes with it after releasing
- Verify if adjusting Spin Middle value that Spin Start and Spin Finish values change with it after releasing
- Verify property values such as Alpha (0 to 1), Horizontal Angle (0 to 180), Vertical Angle (-180 to 180), Pivot (0 to 1), Dimensions (0.001 minimum), Friction (0 to 10), and Density (100 to 10000) stay within their expected range when clicking and dragging on the field
- Verify if clicking on a numerical property to edit the text directly and attempting to set it to values outside of their range that it reverts back to the end of the range after editing
- Verify the numerical property values have step incremental values (the amount the value increases/decreases per each pixel drag) that feel appropriate for the property behavior and the min/max range of the property itself
